cross/bossa/Portfile

```diff
@@ -0,0 +1,51 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           wxWidgets 1.0
+
+github.setup        shumatech BOSSA 1.9.1
+
+name                bossa
+categories          cross devel
+license             BSD
+maintainers         {mit.edu:quentin @quentinmit} \
+                    openmaintainer
+description         Basic Open Source SAM-BA Application
+long_description    BOSSA is a flash programming utility for Atmel's SAM \
+                    family of flash-based ARM microcontrollers.
+
+homepage            http://www.shumatech.com/web/products/bossa
+platforms           darwin
+
+checksums           sha1    ec911282fe3872769a73195c9e0d6e57b2a6e5b2 \
+                    rmd160  729652752baa57dde02ce30608b208ddc753d8c0 \
+                    sha256  811b9c6833cb43072fa3f3e76131fb46cbc36d8f7181b8a25085f0f4ba5ae77d \
+                    size    544207
+
+default_variants    +wxwidgets
+depends_lib-append  port:readline
+
+# Bare Makefile
+use_configure       no
```

Comment:
This means you're not [using the right compiler](https://trac.macports.org/wiki/UsingTheRightCompiler) or -arch flags... Could you try using the makefile 1.0 portgroup and/or making further manual fixes to fix that?

Reply:
I don't understand this comment. The port doesn't have a configure script (upstream just ships a dumb Makefile that only knows how to build, not install). Doesn't that mean I need to specify use_configure no so that it doesn't try to invoke `configure`? I didn't even try the port without use_configure. I could add the makefile portgroup - what does it give me that would be useful with this package?

Reply:
MacPorts knows how to tell ports to use the right compiler and -arch flags when they use an autoconf configure script or cmake or meson or other common build systems. Since this port uses none of those, MacPorts doesn't do that, and it falls to you as portfile maintainer to do it. One method is the makefile portgroup. Read [the comments in its source code](https://github.com/macports/macports-ports/blob/master/_resources/port1.0/group/makefile-1.0.tcl) to find out how to use it.

Reply:
Got it. It looks like the makefile portgroup sets `CXX` and `CXXFLAGS` appropriately, and I need to then patch the Makefile so it doesn't override `-arch`. Here's a compilation command from the latest revision of the PR, please let me know if that's acceptable:

`/usr/bin/clang++  -Wall -Werror -MT obj/CmdOpts.o -MD -MP -MF obj/CmdOpts.d -DVERSION=\"1.9.1\" -g -O2 -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX10.14.sdk -arch x86_64 -stdlib=libc++ -c -o obj/CmdOpts.o src/CmdOpts.cpp`